pkg/cri/resource-manager/policy/builtin/memtier/cache.go

```diff
@@ -55,12 +55,13 @@ func (p *policy) restoreConfig() bool {
 }
 
 type cachedGrant struct {
-	Exclusive string
-	Part      int
-	Container string
-	Pool      string
-	MemType   memoryType
-	Memset    system.IDSet
+	Exclusive   string
+	Part        int
+	Container   string
+	Pool        string
+	MemType     memoryType
+	Memset      system.IDSet
+	MemoryLimit uint64
```

Comment:
*Ideally* the primarily consideration for the fitting algorithm should be requests and not limits. After all, only the requests are supposed to guaranteed for containers.

That said, ATM this is a bit problematic in CRI Resource Manager for memory, because we might not have a good understanding of the memory request at all. When the resource *annotating webhook is not in use* and the container is *Burstable* we don't really know the memory request.